File: app/src/main/java/io/github/xsheeee/cs_controller/SwitchAdapter.java
package io.github.xsheeee.cs_controller;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.materialswitch.MaterialSwitch;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import io.github.xsheeee.cs_controller.Tools.Tools;

public class SwitchAdapter extends RecyclerView.Adapter<SwitchAdapter.SwitchViewHolder> {
    private final List<String> keys;
    private final Map<String, Boolean> configMap;
    private final String configFilePath;
    private final Tools tools;

    // 翻译映射
    private Map<String, String> keyDisplayMap = new HashMap<>();

    public SwitchAdapter(Context context, List<String> keys, Map<String, Boolean> configMap, String configFilePath) {
        this.keys = keys;
        this.configMap = configMap;
        this.configFilePath = configFilePath;
        this.tools = new Tools(context);
    }

    // 提供翻译映射的 setter
    public void setKeyDisplayMap(Map<String, String> keyDisplayMap) {
        this.keyDisplayMap = keyDisplayMap;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public SwitchViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.switch_list_item, parent, false);
        return new SwitchViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull SwitchViewHolder holder, int position) {
        String key = keys.get(position);
        String displayName = keyDisplayMap.getOrDefault(key, key);

        holder.materialSwitch.setText(displayName);

        Boolean isChecked = configMap.get(key);
        if (isChecked != null) {
            holder.materialSwitch.setChecked(isChecked);
            holder.materialSwitch.setEnabled(true);
            holder.materialSwitch.setAlpha(1.0f);
        } else {
            holder.materialSwitch.setEnabled(false);
            holder.materialSwitch.setAlpha(0.5f);
        }

        holder.materialSwitch.setOnCheckedChangeListener((buttonView, isCheckedState) -> {
            configMap.put(key, isCheckedState);
            String newValue = isCheckedState ? "true" : "false";
            tools.updateConfigEntry(configFilePath, key, newValue);
        });
    }

    @Override
    public int getItemCount() {
        return keys.size();
    }

    static class SwitchViewHolder extends RecyclerView.ViewHolder {
        MaterialSwitch materialSwitch;

        public SwitchViewHolder(@NonNull View itemView) {
            super(itemView);
            materialSwitch = itemView.findViewById(R.id.material_switch);
        }
    }
}

File: app/src/main/java/io/github/xsheeee/cs_controller/SwitchAdapter.kt
package io.github.xsheeee.cs_controller

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.materialswitch.MaterialSwitch
import io.github.xsheeee.cs_controller.SwitchAdapter.SwitchViewHolder
import io.github.xsheeee.cs_controller.Tools.Tools

class SwitchAdapter(
    context: Context?,
    private val keys: List<String>,
    private val configMap: MutableMap<String, Boolean>,
    private val configFilePath: String
) :
    RecyclerView.Adapter<SwitchViewHolder>() {
    private val tools = Tools(context)

    // 翻译映射
    private var keyDisplayMap: Map<String, String> = HashMap()

    // 提供翻译映射的 setter
    fun setKeyDisplayMap(keyDisplayMap: Map<String, String>) {
        this.keyDisplayMap = keyDisplayMap
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SwitchViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.switch_list_item, parent, false)
        return SwitchViewHolder(view)
    }

    override fun onBindViewHolder(holder: SwitchViewHolder, position: Int) {
        val key = keys[position]
        val displayName = keyDisplayMap.getOrDefault(key, key)

        holder.materialSwitch.text = displayName

        val isChecked = configMap[key]
        if (isChecked != null) {
            holder.materialSwitch.isChecked = isChecked
            holder.materialSwitch.isEnabled = true
            holder.materialSwitch.alpha = 1.0f
        } else {
            holder.materialSwitch.isEnabled = false
            holder.materialSwitch.alpha = 0.5f
        }

        holder.materialSwitch.setOnCheckedChangeListener { buttonView: CompoundButton?, isCheckedState: Boolean ->
            configMap[key] =
                isCheckedState
            val newValue = if (isCheckedState) "true" else "false"
            tools.updateConfigEntry(configFilePath, key, newValue)
        }
    }

    override fun getItemCount(): Int {
        return keys.size
    }

    class SwitchViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var materialSwitch: MaterialSwitch =
            itemView.findViewById(R.id.material_switch)
    }
}